### ISSUE TITLE
Add more examples with clocked times

### ISSUE THREAD

**Issue by dakra** [alphapapa/org-ql#30]
Hi! First thanks for this great package (and org-super-agenda).

This is not really an issue but I'm unsure where else to ask. Stackoverflow, reddit?

I track all my time in org and `org-ql` sounds nice to query some interesting stats
but I just can't figure out how. Most of it are related to clocked times.
E.g.:

- Show me how many hours I spend per week in a task with a certain tag (e.g. meeting)
- Show me the datetime of the first clock-in and last clock-out per day (per project/file) so I can see when I started working and when I ended.
- Show me all projects (which is a task that has sub-tasks, so the clock from the sub-tasks should be included) where I worked more than X hours on.

As this is not an issue I'll close it in a few days even without an answer (or you can close it right away of course), but I think having more examples is always nice to give people an idea what you can do with it.

Thanks.

**Comment by alphapapa** [comment]
Hi,

Thanks for the kind words.

Yes, I recommend asking general questions on /r/emacs or /r/orgmode first, so other users can benefit from answers.  You can tag `/u/github-alphapapa` if you want my attention on them.

Those would be very interesting examples.  Note that `org-ql` is designed primarily to return a list of heading items, or perform simple actions on them.  So you could use `org-ql` to get a list of tasks with a certain tag, or tasks clocked within certain time periods, but running computations on those tasks and presenting reports on them would be out of `org-ql`'s scope.

Or, from another perspective, `org-ql`, as a framework, can be used with arbitrary predicates.  So if you write a predicate that computes a statistic and returns based on it, you could use that predicate in an  `org-ql` query, because such queries can use any predicate, not just predefined `org-ql` selectors.

To put it another way, if you were thinking of `org-ql` like a full SQL database, I'm afraid it's not.  :)  That would be a fascinating project, to expose Org files with a full, SQL-like API, but I'm guessing that would best be implemented by, e.g. exporting Org files to SQLite databases and then running queries with SQLite.

That's an interesting idea, actually, e.g. an `org-export` SQLite backend...

**Comment by dakra** [comment]
Thanks for your thorough answer.
I'll close this now and make a reddit thread in r/emacs as nobody reads r/orgmode ;)
Thanks.

**Comment by alphapapa** [comment]
> nobody reads r/orgmode ;)

Hey, I do!  I'm also a moderator of it.  ;)
